From 951d190184b36bb6d4c71be42a02cb1e4d3b511a Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 26 Sep 2024 11:32:18 +0300
Subject: [PATCH] [FSW-9512] fs1: increase lfs file name length

Length of max lfs file name was limited by
length of max file name macro from lfs lib.
Due to this there was wrong assumption made
in full file name formation, and corrected
the same by removing one subtraction before
appending file name to file tag.
Increased max file name length to make it
work properly for large file names.

Change-Id: Ieae84b49a172e91fb7541ed5e08e9d5a934c00dc
---

diff --git a/common/littlefs/lfs.h b/common/littlefs/lfs.h
index 50542f6..7f03eb2 100644
--- a/common/littlefs/lfs.h
+++ b/common/littlefs/lfs.h
@@ -49,7 +49,7 @@
 // info struct. Limited to <= 1022. Stored in superblock and must be
 // respected by other littlefs drivers.
 #ifndef LFS_NAME_MAX
-#define LFS_NAME_MAX 10
+#define LFS_NAME_MAX 20
 #endif
 
 // Maximum size of a file in bytes, may be redefined to limit to support other
diff --git a/common/littlefs/lfs_wrapper.c b/common/littlefs/lfs_wrapper.c
index 76a4d20..9a09985 100644
--- a/common/littlefs/lfs_wrapper.c
+++ b/common/littlefs/lfs_wrapper.c
@@ -243,9 +243,6 @@
 		file_name[i] = file_tag[i];
 	}
 
-	if (i > 0)
-		i -= 1;
-
 	for (j = 0; file_num_str[j] && (i < LFS_NAME_MAX); j++, i++) {
 		file_name[i] = file_num_str[j];
 	}
